### ISSUE TITLE
Freezing Whole Device When Chapter Ends

### ISSUE THREAD

**Issue by OutStraw** [Jays2Kings/tachiyomiJ2K#1400]
### Steps to reproduce

At the end of the chapter, from time to time it freezing my whole phone for around 30 seconds. This has happened to me for the past few months, since around April this year. Is there a fix?

### Expected behavior

No freezing at the end of a chapter.

### Actual behavior

Freezing at chapter end occasionally.

### TachiyomiJ2K version

1.5.8

### Android version

Android 10 QP1A.

### Device

Red Mi 9

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I am reporting an issue exclusive to this fork. I have also checked that is not an issue on the [main version of Tachiyomi](https://github.com/tachiyomiorg/tachiyomi)
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[1.5.8](https://github.com/Jays2Kings/tachiyomiJ2K/releases/latest)**.
- [X] I have updated all installed extensions.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I have filled out all of the requested information in this form.

**Comment by KazKaz90** [comment]
What source, manga, manhwa?

**Comment by kyoyacchi** [comment]
It happens to me when i reading a chapter._.

**Comment by e-shl** [comment]
https://github.com/tachiyomiorg/tachiyomi/issues/1908  ???

**Comment by Shamicen** [comment]
If It's at the end of chapters and happens once per hour then it is probably  tachiyomiorg#7847

TLDR: Tachiyomi checks if the next chapter is downloaded once it's 5 pages before a chapter transition and one of the functions that is called during that check is checkRenew(), which will start refreshing the chapter cache if it has been 1 hour since the last refresh. Tachiyomi will be unresponsive as long as the refresh is going on.
